src/Microsoft.DotNet.Wpf/src/Common/Graphics/wgx_core_types.cs

```diff
@@ -454,6 +454,15 @@ internal struct MilColorF
     internal float g;
     internal float b;
     internal float a;
+
+    public override int GetHashCode()
+    {
+        return a.GetHashCode() ^ r.GetHashCode() ^ g.GetHashCode() ^ b.GetHashCode();
+    }
+    public override bool Equals(object obj)

```

Comment:
Because `MilColorF` is tightly-packed, we'll get byte-wise comparison by default. 

Even if there are minor differences in the float values between different instances of the same color value, I'd imagine that `Equals` would return `false`. Do you think this behavior is probably different between .NET Framework and .NET Core due to CoreClr differences, and needs to be remedied with a better implementation for `Equals` - perhaps something like this: 


```C#
public override bool Equals(object obj)
{
    if (Object.ReferenceEquals(this, obj))
    {
        return true;
    }
    
    if (obj is MilColorF color)
    {
        return Equals(color);
    }

    return base.Equals(obj);
}

public bool Equals(MilColorF color)
{
    return DoubleUtil.AreClose(color.r, r) &&
           DoubleUtil.AreClose(color.g, g) && 
           DoubleUtil.AreClose(color.b, b) &&
           DoubleUtil.AreClose(color.a, a);
}
```

/cc @rladuca

Reply:
I don’t think we should do this. This will lead to “equal" colors with unequal hashes. We’ll need to redesign GetHashCode if we wanted to use `AreClose` etc.

Reply:
Normally I'd expect `Equals` to do `color.r == r && color.g == g ...` and not use something like `AreClose`. Bitwise equality might work but it may be confusing at times because it doesn't account for NaNs and other floating point particularities and as implemented now it's probably pretty slow as well. Though it all depends on how theses structs are used...